samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java

```diff
@@ -0,0 +1,680 @@
+      }
+      try {
+        adminClient.close();
+      } catch (Exception e) {
+        LOG.warn("adminClient.close for system " + systemName  + " failed with exception.", e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  KafkaSystemAdmin(String systemName, Supplier<Consumer<K, V>> metadataConsumerSupplier,
+      Supplier<ZkUtils> connectZk, Supplier<AdminClient> connectAdminClient,
+      Map<String, ChangelogInfo> changelogTopicMetaInformation, Map<String, Properties> intermediateStreamProperties,
+      Properties coordinatorStreamProperties, int coordinatorStreamReplicationFactor, boolean deleteCommittedMessages) {
+    this.systemName = systemName;
+    this.metadataConsumer = metadataConsumerSupplier.get();
+    this.adminClient = connectAdminClient.get();
+    this.changelogTopicMetaInformation = changelogTopicMetaInformation;
+    this.intermediateStreamProperties = intermediateStreamProperties;
+    this.coordinatorStreamProperties = coordinatorStreamProperties;
+    this.coordinatorStreamReplicationFactor = coordinatorStreamReplicationFactor;
+    this.connectZk = connectZk;
+    this.connectAdminClient = connectAdminClient;
+    this.deleteCommittedMessages = deleteCommittedMessages;
+  }
+
+  /**
+   * Helper method to create KafkaSystemAdmin
+   * @param systemName system name
+   * @param config application config
+   * @param idPrefix will be used to generate clientId for the metadata consumer
+   * @return new KafkaSystemAdmin object
+   */
+  public static KafkaSystemAdmin getKafkaSystemAdmin(final String systemName, final Config config,
+      final String idPrefix) {
+
+    boolean zkSecure = false; // needs to be added to the argument if ever true is possible
+
+    // populate brokerList from either consumer or producer configs
+    Properties props = new Properties();
+    String brokerListString =
+        config.get(String.format(CONSUMER_CONFIGS_CONFIG_KEY, systemName, ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG));
+    if (brokerListString == null) {
+      brokerListString =
+          config.get(String.format(PRODUCER_CONFIGS_CONFIG_KEY, systemName, ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG));
+    }
+    if (brokerListString == null) {
+      throw new SamzaException(
+          "" + ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG + " is required for systemAdmin for system " + systemName);
+    }
+    props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerListString);
+
+    // kafka.admin.AdminUtils requires zkConnect
+    // this will change after we move to the new org.apache..AdminClient
+    String zkConnectStr = config.get(String.format(CONSUMER_CONFIGS_CONFIG_KEY, systemName, ZOOKEEPER_CONNECT));
+    if (zkConnectStr == null) {
+      throw new SamzaException("Missing zookeeper.connect config for admin for system " + systemName);
+    }
+    props.put(ZOOKEEPER_CONNECT, zkConnectStr);
+
+    Supplier<AdminClient> adminClientSupplier = () -> {
```

Comment:
nit: would prefer them to be consistent with line 200 by replacing it with expression lambda instead of statement.